File: capture.py
import subprocess
with open("python_out.txt", "w", encoding="utf-8") as f:
    try:
        result = subprocess.run(["cargo", "clippy", "--workspace", "--all-targets", "--all-features", "--", "-D", "warnings"], capture_output=True, text=True, check=False)
        f.write(result.stdout)
        f.write("\n--STDERR--\n")
        f.write(result.stderr)
        f.write(f"\nExit code: {result.returncode}")
    except Exception as e:
        f.write(f"Exception: {e}")

File: capture.rs
use std::process::Command;
use std::fs;

fn main() {
    let output = Command::new("cargo")
        .args(&["clippy", "--workspace", "--all-targets", "--all-features", "--message-format=short"])
        .output()
        .expect("Failed to execute command");

    let stdout_str = String::from_utf8_lossy(&output.stdout);
    let stderr_str = String::from_utf8_lossy(&output.stderr);

    let combined = format!("STDOUT:\n{}\n\nSTDERR:\n{}", stdout_str, stderr_str);
    fs::write("capture.log", combined).expect("Unable to write file");
}
